IanGRap/Squish
# Round system

Contributors: Alfredo, Austin

Goals:
- Make a round system with appropriate UI
- Replace game over text with somewhat more involved UI
- Refactor MatchManager

What Changed:
- Imported a new asset called DOTween to easily tween objects
- MatchManager.cs
    - Removed UI controls and random starting config controls
    - Added a field called _currentState which holds the current game state (Play, Pause, MatchOver, GameOver)
    - Exposed some varaibles such as AlivePlayers
- UIMatchManager
    - Moved functionality from MatchManager into this class
    - Now handles the Score screen as well as the Game Over screen 
- UIPauseMenu
    - Removed Restart option
- UIScoreMenu / UIScoreImage
    - Holds all logic for displaying score
- UIGameOverMenu
    - Holds all logic for displaying the GameOver screen
- PlayerColorController
    - Sets colors of score screen icons
- SceneCreator
    - Now contains all logic for creating starting configs
    - Deletes score icons of the players that were not chosen in player select

How to Test:
- Use scene Round System and Round System Player Select
- Play the game with 2 3 and 4 players, ensuring that you see the correct scores for all players
- Ensure that you see the Game Over screen every time someone accumulates 3 wins
- Ensure that you can do a Rematch from the game over screen (this should reset all scores)
- Ensure the pause menu options still work
- Test in a standalone Build!
